packages/strapi-hook-bookshelf/lib/index.js

```diff
@@ -533,14 +533,84 @@ module.exports = function(strapi) {
                       } else if (definition.primaryKeyType !== 'integer') {
                         idAttributeBuilder = [`id ${getType({type: definition.primaryKeyType})} NOT NULL PRIMARY KEY`];
                       }
-                      const columns = generateColumns(attributes, idAttributeBuilder).join(',\n\r');
+                      const columns = generateColumns(attributes, idAttributeBuilder).join(', ');
+
+                      // FIXME: For temporary debugging. This should be eliminated.
+                      const orgQuery = `CREATE TABLE ${quote}${table}${quote} (${columns})`;
+
+                      const newQuery = ORM.knex.schema.createTable(table, dbTable => {
+
+                        if (definition.primaryKeyType === 'integer') {
+                          // Common case, numeric incrementable key
+                          dbTable.increments();
+
+                        } else {
+                          // Build primary key with a peculiar field type
+                          const pk = dbTable[definition.primaryKeyType]('id')
+                            .primary()
+                            .notNullable();
+
+                          if (definition.client === 'pg' && definition.primaryKeyType === 'uuid') {
+                            pk.defaultTo(ORM.knex.raw('uuid_generate_v4()'));
+                          }
+                        }
+
+                        _.mapKeys(attributes, (field, fieldName) => {
+                          try {
+                            if (!field.type) {
+                              // Creating integer column for foreign key as the original implementation
+                              // ROOM FOR IMPROVEMENT -- Use actual foreign keys
+                              field.model && dbTable.integer(field.model);
+                              return;
+                            }
+
+                            switch(field.type) {
+                              // ROOM FOR IMPROVEMENT -- We could add REAL defined constrains to database
+                              case 'email':
+                              case 'password':
```

Comment:
password isn't be hidden by default

Reply:
Do you mean in the GUI? ... or not been hashed in the DB?
If is something of that, should not be caused by this changes. The same behavior should be present in `master`. If not, can you expand what are you meaning please?